spec/factories/loans.rb

```diff
@@ -55,11 +55,12 @@
 
 
     trait :active do
-      status_option_id Loan::STATUS_OPTIONS.value_for('Active')
+      # note: can't seem to count on the seed.db data having been loaded when these are executed
+      status_option_id 1 #Loan.status_option_set.value_for_label('Active')
```

Comment:
I talked to Tom and he agrees that it’s bad practice for tests to rely on seed data. He says he’s never seen anyone do that, and it’s likely to lead to problems that could be very hard to debug. I'm curious about what bugs you say you've been encountering due to *not* using the seed data—it seems more likely to me that using it that would cause bugs. Just as one example, if you're only seeding the db once before running the test suite, and then someday you come across a test that needs to use truncate rather than transaction, the tests that run after that one may not have all the seed data. That sounds like a nightmare to debug. 

Using factories is a best practice, and it shouldn't take so much more time if we design it right. Mel's code sample is a good start, but as written it would create an additional 'active' status in the db each time an additional loan was created within the same spec. Tom suggested using a pattern similar to elmo, where at the top of the missions factory is a method that creates a new mission only if one isn't already there:

    def get_mission
      Mission.first || create(:mission)
    end

Then in a factory that needs an associated mission, `mission { get_mission }`.

That method is used all over the specs. In that case, it's only important that there be one mission. In our case it would likely need to specify certain attributes, but the same principle holds. Maybe something like this:

    def get_loan_status(status)
      Option.where(value: status) || create(:option, value: status)
    end

Using the simplified translation system Mel suggested would also remove some of the need for depending on database content, thankfully. 

So, let's please consider this settled and move on. Plan to do whatever it takes to avoid relying on seed data in the tests. Joseph, if you're ever having trouble figuring out the best way to generate the data you need (or just want to commiserate about how much generating fake data sucks), I or I'm sure Mel would be happy to help. 

I hope there are no hard feelings. You're both doing great work, you both made good arguments, and the answer here was not obvious. This kind of debate makes our products better (and me smarter :). 

On a related note, I think I've also come around to agree with both of you that certain options, like 'active', are necessary business logic and should be hardcoded in, and that we should use slugs to reference them. I suggest we put a `system` boolean field on the options table to signify options that the code relies on and thus should never be deleted.